src/nifake/unit_tests/test_library_interpreter.py

```diff
@@ -843,6 +843,49 @@ def test_import_attribute_configuration_buffer_str(self):
             interpreter.import_attribute_configuration_buffer(configuration)
         self.patched_library.niFake_ImportAttributeConfigurationBuffer.assert_called_once_with(_matchers.ViSessionMatcher(SESSION_NUM_FOR_TEST), _matchers.ViInt32Matcher(len(configuration)), _matchers.ViInt8BufferMatcher(expected_list))
 
+    def test_write_waveform_numpy_complex128_valid_input(self):
+        import ctypes
+        import numpy as np
```

Comment:
ctypes and numpy are already imported at the top of the module. You don't need to import them again, here.